skins/cat17/src/scripts/donationForm.js

```diff
@@ -78,6 +78,15 @@ $( function () {
 			viewHandler: WMDE.View.createElementClassSwitcher( $( '#type-donor .wrap-field.anonym .wrap-input' ), /BEZ/, 'disabled' ),
 			stateKey: 'donationFormContent.paymentType'
 		},
+		// Disable recurring intervals when Sofort is selected
+		{
+			viewHandler: WMDE.View.createElementClassSwitcher( $( '#recurrence .wrap-field.recurring .wrap-input' ), /SUB/, 'disabled' ),
+			stateKey: 'donationFormContent.paymentType'
+		},
+		{
+			viewHandler: WMDE.View.createElementClassSwitcher( $( '#payment-method .wrap-field.sofort .wrap-input' ), /[^0]+/, 'disabled' ),
```

Comment:
This whole part of the code is some seriously weird territory, but the "not 0" regex makes me look like I am playing chess while the rest of the world is still playing checkers 🎓:

`/[^0]+/`

It seems to work fine, though!

Reply:
I think this is the shortest regexp that checks for `not 0`. Anything else will be less readable.

Reply:
The regex is fine and the ElementClassSwitcher works as intended. 

The only "better" way to do this is converting all those ViewHandlers to Vue.